File: Step4_compute_scores.py
import argparse
import json


def compute_scores(path):
    with open(path, "r", encoding="utf-8") as f:
        data = json.load(f)

    scores = {}
    for dataset_name, rows in data.items():
        total = len(rows)
        correct = sum(1 for item in rows.values() if item.get("correct"))
        by_task = {}
        for item in rows.values():
            task = item.get("task_type", "unknown")
            by_task.setdefault(task, {"total": 0, "correct": 0})
            by_task[task]["total"] += 1
            by_task[task]["correct"] += int(bool(item.get("correct")))
        for task, counts in by_task.items():
            counts["accuracy"] = counts["correct"] / counts["total"] if counts["total"] else 0
        scores[dataset_name] = {
            "total": total,
            "correct": correct,
            "accuracy": correct / total if total else 0,
            "by_task": by_task,
        }
    return scores


if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("--merge_file", type=str, default="./ChronoPhyBench_Input.json")
    parser.add_argument("--score_output_file", type=str, default="./ChronoPhyBench_scores.json")
    args = parser.parse_args()

    scores = compute_scores(args.merge_file)
    with open(args.score_output_file, "w", encoding="utf-8") as f:
        json.dump(scores, f, indent=2, ensure_ascii=False)
    print(json.dumps(scores, indent=2, ensure_ascii=False))

File: scripts/step4_compute_scores.mjs
import { mkdir, readFile, writeFile } from "node:fs/promises";
import path from "node:path";

function parseArgs(argv) {
  const args = {
    mergeFile: path.resolve("ChronoPhyBench_Input.json"),
    scoreOutputFile: path.resolve("ChronoPhyBench_scores.json"),
  };
  for (let i = 0; i < argv.length; i += 1) {
    const key = argv[i];
    const value = argv[i + 1];
    if (key === "--merge_file") args.mergeFile = path.resolve(value);
    if (key === "--score_output_file") args.scoreOutputFile = path.resolve(value);
    if (key.startsWith("--")) i += 1;
  }
  return args;
}

const args = parseArgs(process.argv.slice(2));
const data = JSON.parse(await readFile(args.mergeFile, "utf8"));
const scores = {};

for (const [datasetName, rows] of Object.entries(data)) {
  const values = Object.values(rows);
  const total = values.length;
  const correct = values.filter((item) => item.correct).length;
  const byTask = {};
  for (const item of values) {
    const task = item.task_type ?? "unknown";
    byTask[task] ??= { total: 0, correct: 0 };
    byTask[task].total += 1;
    byTask[task].correct += item.correct ? 1 : 0;
  }
  for (const counts of Object.values(byTask)) {
    counts.accuracy = counts.total ? counts.correct / counts.total : 0;
  }
  scores[datasetName] = {
    total,
    correct,
    accuracy: total ? correct / total : 0,
    by_task: byTask,
  };
}

const payload = `${JSON.stringify(scores, null, 2)}\n`;
await mkdir(path.dirname(args.scoreOutputFile), { recursive: true });
await writeFile(args.scoreOutputFile, payload, "utf8");
console.log(payload);
